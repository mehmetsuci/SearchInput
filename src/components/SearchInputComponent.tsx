import React from 'react';

interface SearchInputComponentProps {
  handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  runSearch: () => void;
  changeSearch: () => void;
  buttonStyle: React.CSSProperties;
  buttonIconClass: string;
  placeholder: string;
  tabIndex: number | undefined;
  keyword: string;
}

const SearchInputComponent = ({
  handleInputChange,
  runSearch,
  changeSearch,
  buttonStyle,
  buttonIconClass,
  placeholder,
  tabIndex,
  keyword,
}: SearchInputComponentProps) => {
  const btnClass = `btn${buttonStyle ? ' btn-' + buttonStyle : ''}`;
  const btnIconClass = `glyphicon${
    buttonIconClass ? ' glyphicon-' + buttonIconClass : ''
  }`;
  return (
    <div className="input-group searchInput">
      <input
        type="text"
        className="form-control"
        onChange={handleInputChange}
        onBlur={changeSearch}
        onKeyDown={e => {
          if (e.key === 'Enter') runSearch();
        }}
        placeholder={placeholder}
        tabIndex={tabIndex}
        value={keyword}
      />
      <span className="input-group-btn searchInputButton">
        <button
          className={btnClass}
          onClick={runSearch}
          type="button"
          tabIndex={tabIndex}
        >
          <span className={btnIconClass}></span>
        </button>
      </span>
    </div>
  );
};

export default SearchInputComponent;
