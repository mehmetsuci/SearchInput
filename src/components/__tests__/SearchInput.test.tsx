import { render } from '@testing-library/react';
import React from 'react';

import parseStyle from '../../utils/parseStyle';
import SearchInputComponent from '../SearchInputComponent';

describe('SearchInput Component Test Unit', () => {
  it('should render correctly', () => {
    const props = {
      keyword: 'something',
      runSearch: () => {},
      changeSearch: () => {},
      handleInputChange: () => {},
      className: 'awsomeClass',
      tabIndex: 2,
      style: parseStyle(
        'border-style:solid;border-color:#39c;border-width:5px;'
      ),
      buttonStyle: undefined,
      buttonIconClass: '',
      placeholder: '',
    };

    const { container } = render(<SearchInputComponent {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
