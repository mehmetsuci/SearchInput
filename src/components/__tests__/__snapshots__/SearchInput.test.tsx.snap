// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component Test Unit should render correctly 1`] = `
<div
  class="input-group searchInput"
>
  <input
    class="form-control"
    placeholder=""
    tabindex="2"
    type="text"
    value="something"
  />
  <span
    class="input-group-btn searchInputButton"
  >
    <button
      class="btn"
      tabindex="2"
      type="button"
    >
      <span
        class="glyphicon"
      />
    </button>
  </span>
</div>
`;
